apiserver/facades/client/client/client.go

```diff
@@ -792,14 +812,127 @@ func (c *Client) AbortCurrentUpgrade() error {
 }
 
 // FindTools returns a List containing all tools matching the given parameters.
-func (c *Client) FindTools(args params.FindToolsParams) (params.FindToolsResult, error) {
+func (c *ClientV3) FindTools(args params.FindToolsParams) (params.FindToolsResult, error) {
+	logger.Criticalf("ClientV3.FindTools args %#v", args)
 	if err := c.checkCanWrite(); err != nil {
 		return params.FindToolsResult{}, err
 	}
-
 	return c.api.toolsFinder.FindTools(args)
 }
 
+// FindTools returns a List containing all tools matching the given parameters.
+func (c *Client) FindTools(args params.FindToolsParams) (result params.FindToolsResult, err error) {
+	defer func() {
+		logger.Criticalf("Client.FindTools args -> %#v, params.FindToolsResult -> %#v, error -> %#v", args, result, err)
+	}()
+
+	if err := c.checkCanWrite(); err != nil {
+		return params.FindToolsResult{}, err
+	}
+	result, err = c.api.toolsFinder.FindTools(args)
+	if err == nil {
```

Comment:
combine these two expressions into
`err == nil && result.Error != nil`

Reply:
Yeah, it's been changed, and this kinda error translation has been moved to API client-side.
Just not pushed yet.